src/views/TimelineTrackView.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import {PauseIcon, PlayIcon} from "../icons";
+import {ChangeEvent, MouseEvent, useCallback} from "react";
+
+interface Props {
+    frameStart: number,
+    frame: number, setFrame: React.Dispatch<React.SetStateAction<number>>,
+    frameEnd: number,
+
+    frameCropStart: number, setCropStart: React.Dispatch<React.SetStateAction<number>>,
+    frameCropEnd: number, setCropEnd: React.Dispatch<React.SetStateAction<number>>,
+
+    frameRef: React.MutableRefObject<number>,
+
+    playing: boolean, setPlaying: React.Dispatch<React.SetStateAction<boolean>>,
+    loopPlayback: boolean, setLoopPlayback: React.Dispatch<React.SetStateAction<boolean>>,
+}
+
+export default function TimelineTrackView(
+    {
+        frameStart,
+        frame, setFrame,
+        frameEnd,
+
+        frameCropStart, setCropStart,
+        frameCropEnd, setCropEnd,
+
+        frameRef,
+
+        playing, setPlaying,
+        loopPlayback, setLoopPlayback,
+    }: Props
+) {
+    /* Play button (on click) */
+    const togglePlaying = useCallback(() => setPlaying(current => {
+        /* Handle start from pause on last frame, if encountered */
+        if (frameRef.current>=frameCropEnd) setFrame(frameCropStart);
+        /* Invert current playing status */
+        return !current;
+    }), [frameRef, frameCropStart, frameCropEnd]);
+
+    /* Play button loop checkbox (on change) */
+    const toggleLooping = useCallback(({target: {checked}}: ChangeEvent<HTMLInputElement>) => {
+        setLoopPlayback(checked);
+    }, []);
+
+    /* Timeline track thumb (on change) */
+    const seek = useCallback(({target: {value}}: ChangeEvent<HTMLInputElement>) => {
+        const thumbVal = parseInt(value);
+        /* ThumbVal is in range: set frame to it */
+        if (frameCropStart<=thumbVal && thumbVal<=frameCropEnd)
+            setFrame(thumbVal);
+        /* ThumbVal is too low: decrement frame until at crop start */
+        else if (thumbVal<frameCropStart && frameCropStart<=frameRef.current-1)
+            setFrame(frameRef.current-1);
+        /* ThumbVal is too high: increment frame until at crop end */
+        else if (thumbVal>frameCropEnd && frameCropEnd>=frameRef.current+1)
+            setFrame(frameRef.current+1);
+    }, [frameRef, frameCropStart, frameCropEnd]);
+
+    /* Timeline track (on context menu) */
+    const resetCrop = useCallback((e: MouseEvent<HTMLInputElement>) => {
+        e.preventDefault();
+        if (e.button==2) {
```

Comment:
should add at least a comment explaining which button this is

Reply:
but also why do you need to check the button? shouldn't onContextMenu already be doing the right thing?

Reply:
I may have originally written the callback first, for which the event type isn't explicit about being a context menu event, so I didn't see the redundancy of the mouse button number check. Fixed.